envoyextproc/evaluation.go

```diff
@@ -0,0 +1,159 @@
+package envoyextproc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-policy-agent/opa/v1/ast"
+	"github.com/open-policy-agent/opa/v1/bundle"
+	"github.com/open-policy-agent/opa/v1/config"
+	"github.com/open-policy-agent/opa/v1/logging"
+	"github.com/open-policy-agent/opa/v1/rego"
+	"github.com/open-policy-agent/opa/v1/storage"
+	"github.com/open-policy-agent/opa/v1/topdown/builtins"
+	iCache "github.com/open-policy-agent/opa/v1/topdown/cache"
+	"github.com/open-policy-agent/opa/v1/topdown/print"
+	"github.com/open-policy-agent/opa/v1/tracing"
+)
+
+// EvalContext - This is an SPI that has to be provided if the envoy external authorization
+// is used from outside the plugin, i.e. as a Go module
+type EvalContext interface {
```

Comment:
This interface is identical to the one in `envoyauth`. Let's refactor this, and the `PrepareQueryOpts` type below into a single interface in a shared package and alias the one in `envoyauth` to refer to the shared package.